code/game/gamemodes/cult/ritual.dm

```diff
@@ -239,24 +232,28 @@ This file contains the arcane tome files as well as innate cultist emergency com
 			break
 	if(!rune_to_scribe)
 		return
+	var/turf/Thenewturfyouwalkedto = get_turf(user) //we may have moved. adjust as needed...
+	if(locate(/obj/effect/rune) in Thenewturfyouwalkedto)
+		user << "<span class='cult'>There is already a rune here.</span>"
+		return
 	if(!Adjacent(user) || !src || qdeleted(src) || user.incapacitated())
 		return
 	user.visible_message("<span class='warning'>[user] cuts open their arm and begins writing in their own blood!</span>", \
 						 "<span class='cult'>You slice open your arm and begin drawing a sigil of the Geometer.</span>")
 	if(iscarbon(user))
 		var/mob/living/carbon/C = user
-		C.apply_damage(0.1, BRUTE, pick("l_arm", "r_arm"))
-		if("Call Forth The Geometer" == entered_rune_name)
-			C.apply_damage(40, BRUTE, pick("l_arm", "r_arm"))
+		C.apply_damage(initial(rune_to_scribe.scribe_damage), BRUTE, pick("l_arm", "r_arm"))
+		if(ispath(rune_to_scribe, /obj/effect/rune/narsie))
 			var/area/A = get_area(src)
 			var/locname = initial(A.name)
-			priority_announce("Figments from an eldritch god are being summoned by [user] into [locname] from an unknown dimension. Disrupt the ritual before it reaches a critical point.","Central Command Higher Dimensionsal Affairs")
-			if(!do_after(user, 500, target = get_turf(user)))
-				return
-	if(!do_after(user, 50, target = get_turf(user)))
+			priority_announce("Figments from an eldritch god are being summoned by [user] into [locname] from an unknown dimension. Disrupt the ritual at all costs.","Central Command Higher Dimensionsal Affairs")
+	if(!do_after(user, initial(rune_to_scribe.creation_delay), target = Thenewturfyouwalkedto))
```

Comment:
From checking the code it appears I need a target for a progress par to appear; I'll test and see if it works without one but I think it does need it.